packages/cli/src/ui/commands/types.ts

```diff
@@ -106,12 +106,54 @@ export type SlashCommandActionReturn =
   | OpenDialogActionReturn
   | LoadHistoryActionReturn;
 
+/**
+ * Metadata describing a command's origin and expected behavior.
+ */
+export interface SlashCommandMetadata {
+  /** The source of the command definition. */
+  source: 'built-in' | 'file';
```

Comment:
Instead of source we may want to utilize a `kind` type here. Primary reason if we have more than 1 file type we support in the future this will become ambiguous